projects/electron-ngrx/src/lib/selector-manager.ts

```diff
@@ -0,0 +1,22 @@
+import {h64} from 'xxhashjs';
+
+type fooType = (...args: any) => any;
+const hashFunction = (foo: fooType, identifier: string): string => {
+  return h64(foo.name + foo.toString() + identifier, 0xABCD).toString(16);
+};
+let selectorStorage: Record<string, fooType> = {};
+export const registerSelector = (selector, identifier = '') => {
+  const hash = hashFunction(selector, identifier);
+  if (selectorStorage.hasOwnProperty(hash)) {
+    throw new Error(`duplicate selector.\
+     the selector: '${selector.name}' Already exist.\
+      if you didnt register this selector Already, Maybe you register another selector with the same name.\
+       in this case consider to use the second argument in registerSelector as identifier of the selector`);
+  }
+  selectorStorage[hash] = selector;
+  (selector as any).hash = hash;
```

Comment:
If possible - create a type for the selector that has this property as an optional